pandas/core/arrays/arrow/array.py

```diff
@@ -493,7 +493,21 @@ def _box_pa_array(
             if pa.types.is_dictionary(pa_type):
                 pa_array = pa_array.dictionary_encode()
             else:
-                pa_array = pa_array.cast(pa_type)
+                try:
+                    pa_array = pa_array.cast(pa_type)
+                except (
+                    pa.ArrowInvalid,
+                    pa.ArrowTypeError,
+                    pa.ArrowNotImplementedError,
+                ):
+                    if pa.types.is_string(pa_array.type) or pa.types.is_large_string(
+                        pa_array.type
+                    ):
+                        return cls._from_sequence_of_strings(
+                            value, dtype=pa_type
```

Comment:
Looking at _from_sequence_of_strings, this seems like it would make a lot more sense for the implementation to live in _from_sequence and that to call this rather than vice-versa (likely for all dtypes not just time, but haven't looked that closely)

That said i dont think there's anything wrong with this implementation

Reply:
Yeah I think even further the logic of `_from_sequence_of_strings` should be in `_box_pa_array` (which is used in fillna, comparison ops) which can be done in a followup